docs/performance/sharding.md

```diff
@@ -1,41 +1,133 @@
+Applying sharding can drastically improve the performance on large datasets.
+:::
 
-A brief recap: CrateDB tables are split into a configured number of shards, and
-then these shards are distributed across the cluster.
+This document is a sharding best practice guide for CrateDB.
+A brief recap: CrateDB tables are split into a configured number of shards.
+These shards are distributed across the cluster to optimize concurrent and
+parallel data processing.
 
-Figuring out how many shards to use for your tables requires you to think about
-the type of data you're processing, the types of queries you're running, and
-the type of hardware you're using.
+Whenever possible, CrateDB will parallelize query workloads and distribute them
+across the whole cluster. The more CPUs this query workload can be distributed
+across, the faster the query will run.
 
-:::{NOTE}
+:::{seealso}
 This guide assumes you know the basics.
+If you are looking for an intro to sharding, see also the
+{ref}`sharding-partitioning` and the
+{ref}`sharding reference <crate-reference:ddl-sharding>` documentation.
+:::
+
+
+## General recommendations
+
+To avoid running your clusters with too many shards or too large shards,
+implement the following guidelines as a rule of thumb:
+
+- Use shard sizes between 10GB and 50GB.
+
+- Keep the number of records on each shard below 200 million.
+
+
+## Sizing considerations
+
+General principles requires careful consideration for cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
 
-If you are looking for an intro to sharding, see {ref}`sharding
-<crate-reference:ddl-sharding>`.
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 3-70 GB of data, with 10-50 GB being the sweet spot for most
+workloads. In large clusters, this often means fewer shards than total CPU cores,
+as larger shards can still be processed efficiently by multiple CPU cores during
+query execution.
+Smaller shards also result in reduced Lucene index efficiency, which can adversely
+affect computed search term relevance.
+
+### CPU-to-shard ratio
+
+If most nodes have more shards per table than they have CPUs, the cluster can
+experience performance degradations.
+For example, on clusters with substantial CPU resources (e.g., 8 nodes × 32 CPUs
+= 256 total CPUs), creating 256+ shards per table often proves counterproductive.
+If you don't manually set the number of shards per table, CrateDB will make a
+best guess, based on the assumption that your nodes have two CPUs each.
+
+### 1000 shards per node limit
+
+To avoid _oversharding_, CrateDB by default limits the number of shards per node to
+1_000 as a critical stability boundary. Any operation that would exceed that limit
+leads to an exception.
```

Comment:
@seut mentioned about this:

> Actually, it's more of a protection than a ‘critical stability boundary’ in my view, as it depends on the environment, i.e. what kind of hardware I'm running.

Reply:
Elsewhere, it is called "soft limit"? Shall we use that term?

> [...] there is a soft limit of 1000 shards per node;

Reply:
I think `protection limit` is much more helpful than `soft limit`. Otherwise I'd question what this `soft limit` actually mean.